Source/core/WarningReportingCategories.h

```diff
@@ -80,5 +80,65 @@ namespace WarningReporting {
         static constexpr uint32_t DefaultWarningBound = { 0 };
         static constexpr uint32_t DefaultReportBound = { 0 };
     };
+
+    class EXTERNAL JobTooLongToFinish {
+    public:
+        JobTooLongToFinish(const JobTooLongToFinish&) = delete;
+        JobTooLongToFinish& operator=(const JobTooLongToFinish&) = delete;
+        JobTooLongToFinish() = default;
+        ~JobTooLongToFinish() = default;
+
+        //nothing to serialize/deserialize here
+        uint16_t Serialize(uint8_t[], const uint16_t) const
+        {
+            return 0;
+        }
+
+        uint16_t Deserialize(const uint8_t[], const uint16_t)
+        {
+            return 0;
+        }
+
+        void ToString(string& visitor, const int64_t actualValue, const int64_t maxValue) const
+        {
+            visitor = (_T("Job took too long to finish"));
+            visitor += Core::Format(_T(", value %lld [ms], max allowed %lld [ms]"), actualValue, maxValue);
+        };
+
+        static constexpr uint32_t DefaultWarningBound = { 7500 };
+        static constexpr uint32_t DefaultReportBound = { 5000 };
+    };
+
+
+    class EXTERNAL JobTooLongWaitingInQueue {
+    public:
+        JobTooLongWaitingInQueue(const JobTooLongWaitingInQueue&) = delete;
+        JobTooLongWaitingInQueue& operator=(const JobTooLongWaitingInQueue&) = delete;
+        JobTooLongWaitingInQueue() = default;
+        ~JobTooLongWaitingInQueue() = default;
+
+        //nothing to serialize/deserialize here
+        uint16_t Serialize(uint8_t[], const uint16_t) const
+        {
+            return 0;
```

Comment:
Don't you want so serialice/deserialize _timeInMs?

Reply:
_timeInMs was leftover from the previous implementation. It is unused now, because this category is supposed to use in REPORT_OUTOFBOUND_CATEGORY macro, and elapsed time is passed there.